azurerm/resource_arm_sql_database.go

```diff
@@ -237,6 +285,27 @@ func resourceArmSqlDatabaseCreateUpdate(d *schema.ResourceData, meta interface{}
 		return err
 	}
 
+	if _, ok := d.GetOk("database_import"); ok {
+		if !strings.EqualFold(createMode, "default") {
+			return fmt.Errorf("database_import can only be used when create_mode is Default")
```

Comment:
@hbuckle Should this error be a validation on the `database_import` property instead? Otherwise, you could be running your scripts and end up with the error during a deployment instead of when generating the plan? Or possibly use `ConflictsWith: ["create_mode"]` on the `database_import` since the default for `create_mode` is `default` (and its required to be default).

Reply:
@vlouwagie ideally yes, however unfortunately the validation API's only expose a single property at this point; as such unfortunately I don't think that's feasible at this time